Fix an uncaught error / Add retry button

m600x opened the pull request:
When trying to get back to the previous question from the first one, an uncaught error is displayed
in the console because the onChange send a `None` which is not accepted. Changed to `Some(0.0)` solve the issue.

STR:
- Click on `Question précédente`
- See console.

```
index.e6e56be7b0c8212e7ab1.js:22 Uncaught {RE_EXN_ID: "Failure", _1: "tl", Error: Error
    at https://darons.io/index.e6e56be7b0c8212e7ab1.js:45:82087
    at onChange (https://daro…}Error: Error
    at https://darons.io/index.e6e56be7b0c8212e7ab1.js:45:82087
    at onChange (https://darons.io/index.e6e56be7b0c8212e7ab1.js:45:82098)
    at Fi (https://darons.io/index.e6e56be7b0c8212e7ab1.js:45:57975)
    at onClick (https://darons.io/index.e6e56be7b0c8212e7ab1.js:45:79571)
    at Object.l (https://darons.io/index.e6e56be7b0c8212e7ab1.js:22:455)
    at p (https://darons.io/index.e6e56be7b0c8212e7ab1.js:22:598)
    at https://darons.io/index.e6e56be7b0c8212e7ab1.js:22:744
    at y (https://darons.io/index.e6e56be7b0c8212e7ab1.js:22:830)
    at at (https://darons.io/index.e6e56be7b0c8212e7ab1.js:22:16358)
    at it (https://darons.io/index.e6e56be7b0c8212e7ab1.js:22:16168)RE_EXN_ID: "Failure"_1: "tl"__proto__: Object
lt @ index.e6e56be7b0c8212e7ab1.js:22
pt @ index.e6e56be7b0c8212e7ab1.js:22
L @ index.e6e56be7b0c8212e7ab1.js:22
W @ index.e6e56be7b0c8212e7ab1.js:22
Jt @ index.e6e56be7b0c8212e7ab1.js:22
Gt @ index.e6e56be7b0c8212e7ab1.js:22
t.unstable_runWithPriority @ index.e6e56be7b0c8212e7ab1.js:30
Vi @ index.e6e56be7b0c8212e7ab1.js:22
A @ index.e6e56be7b0c8212e7ab1.js:22
Xt @ index.e6e56be7b0c8212e7ab1.js:22
```

m600x committed:
Fix an uncaught error
When trying to get back to the previous question from the first one, an uncaught error is displayed
in the console cause the onChange send a None which is not accepted. Changed to Some(0.0) solve the issue.

keuhdall commented:
Hello Anthony !

It is indeed a small mistake I missed !
However I would proceed otherwise with the fix, as doing things this way does not fix the unsafe behavior in the `onChange` function.
A better fix imo would be changing the pattern matching in `Quiz.re` from:

```reason
    let newAnswers = switch answer {
    | Some(value) => [value, ...state.answers]
    | None        => List.tl(state.answers)
    };
```
To:


```reason
    let newAnswers = switch answer {
    | Some(value)                   => [value, ...state.answers]
    | None when state.answers != [] => List.tl(state.answers)
    | _                             => []
    };
```

m600x commented:
Hello Léonard,

Honestly I've no idea what I'm doing (I don't even know which language is used) so I do believe you on the fact that your solution is handling the issue in a much more elegant manner than mine.

Regards,
Anthony

m600x committed:
Uncaught error and retry button
- Change the way an uncaught error is handled
- Add a button to retry at the test at the end

m600x commented:
Your suggestion has been implemented and I also added a retry button on the result page for people that are shamed by their results so they can redo the test and fake the result to show off to the world that they aren't `that old`.

keuhdall commented:
LGTM, thanks for contributing ! 🎉
